<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SerialPortChat</class>
 <widget class="QWidget" name="SerialPortChat">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>480</width>
    <height>468</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout" stretch="2,0,0,0">
   <property name="spacing">
    <number>1</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QListWidget" name="chatList">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="contextMenuPolicy">
      <enum>Qt::CustomContextMenu</enum>
     </property>
     <property name="styleSheet">
      <string notr="true">border: none;</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="horizontalScrollBarPolicy">
      <enum>Qt::ScrollBarAlwaysOff</enum>
     </property>
     <property name="verticalScrollMode">
      <enum>QAbstractItemView::ScrollPerPixel</enum>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="leftMargin">
      <number>9</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>9</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QTextEdit" name="sendBox">
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>150</height>
        </size>
       </property>
       <property name="horizontalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOff</enum>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <property name="spacing">
      <number>6</number>
     </property>
     <property name="leftMargin">
      <number>9</number>
     </property>
     <property name="topMargin">
      <number>9</number>
     </property>
     <property name="rightMargin">
      <number>9</number>
     </property>
     <property name="bottomMargin">
      <number>3</number>
     </property>
     <item>
      <widget class="QCheckBox" name="intervalSend">
       <property name="toolTip">
        <string>&lt;style&gt;
* {
    font-family: &quot;Segoe UI&quot;, &quot;微软雅黑&quot;, sans-serif;
    font-weight: bold;
}
h2 {
    color: red;
}
&lt;/style&gt;
&lt;h2&gt;警告：&lt;/h2&gt;
&lt;p&gt;如果将定时的间隔时间设置得过短致使定时发送按钮无法取消勾选，请立即迅速清空发送框的所有文本，然后取消勾选本项！&lt;/p&gt;</string>
       </property>
       <property name="text">
        <string>定时发送</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="intervalSendValue">
       <property name="suffix">
        <string> ms</string>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>3600000</number>
       </property>
       <property name="value">
        <number>250</number>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QCheckBox" name="autoClearBtn">
       <property name="toolTip">
        <string>开启后，每发送一条消息，都会自动清空发送框的文本</string>
       </property>
       <property name="text">
        <string>发送后自动清空</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <property name="spacing">
      <number>6</number>
     </property>
     <property name="leftMargin">
      <number>9</number>
     </property>
     <property name="topMargin">
      <number>3</number>
     </property>
     <property name="rightMargin">
      <number>9</number>
     </property>
     <property name="bottomMargin">
      <number>8</number>
     </property>
     <item>
      <widget class="QRadioButton" name="sendUtf8">
       <property name="toolTip">
        <string>以 UTF-8 编码发送文本</string>
       </property>
       <property name="text">
        <string>UTF-8</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="sendGbk">
       <property name="toolTip">
        <string>以 GBK 编码发送文本</string>
       </property>
       <property name="text">
        <string>GBK</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="sendHex">
       <property name="toolTip">
        <string>以 16 进制编码发送数据</string>
       </property>
       <property name="text">
        <string>HEX</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="sendNewLine">
       <property name="toolTip">
        <string>每发送一个数据，自动在末尾添加回车换行符 (\r\n)（如果没有）</string>
       </property>
       <property name="text">
        <string>新行</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="clearSend">
       <property name="minimumSize">
        <size>
         <width>80</width>
         <height>0</height>
        </size>
       </property>
       <property name="text">
        <string>清空</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="sendBtn">
       <property name="minimumSize">
        <size>
         <width>80</width>
         <height>0</height>
        </size>
       </property>
       <property name="toolTip">
        <string>按 Ctrl + Enter 键发送</string>
       </property>
       <property name="text">
        <string>发送</string>
       </property>
       <property name="shortcut">
        <string>Ctrl+Return</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>chatList</tabstop>
  <tabstop>sendBox</tabstop>
  <tabstop>sendBtn</tabstop>
  <tabstop>sendHex</tabstop>
  <tabstop>clearSend</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>clearSend</sender>
   <signal>clicked()</signal>
   <receiver>sendBox</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>344</x>
     <y>449</y>
    </hint>
    <hint type="destinationlabel">
     <x>239</x>
     <y>323</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
